angular/projects/admin-nrpti/src/app/communications/communications.component.ts

```diff
@@ -0,0 +1,113 @@
+import { Component, OnInit, OnDestroy, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs/Subject';
+import { FormGroup, FormControl } from '@angular/forms';
+import { CommunicationsPackage } from '../../../../common/src/app/models/master/common-models/communications-package';
+import { FactoryService } from '../services/factory.service';
+import { DatePickerComponent, LoadingScreenService, Utils } from 'nrpti-angular-components';
+
+@Component({
+  selector: 'communications-add',
+  templateUrl: './communications.component.html',
+  styleUrls: ['./communications.component.scss']
+})
+export class CommunicationsComponent implements OnInit, OnDestroy {
+  @ViewChild(DatePickerComponent) DatePicker: DatePickerComponent;
+
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public myForm: FormGroup;
+
+  public commPackage: CommunicationsPackage;
+  public selectedApplication: string;
+
+  constructor(
+    public route: ActivatedRoute,
+    public router: Router,
+    private factoryService: FactoryService,
+    private utils: Utils,
+    private loadingScreenService: LoadingScreenService
+  ) {}
+
+  ngOnInit() {
+    this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
+      this.commPackage = res.communicationsPackage.COMMUNICATIONS;
+      const urlSegments = this.router.url.split('/');
```

Comment:
In the route it is `/communications/:application` so there should be a way to access the `:application` URL param rather than splitting it out here.  I know there is a `paramMap` and a `route.snapshot` that I believe can access it.  This might be overkill though.

Reply:
Nope, just garbage code put in before doing the routing that I forgot to replace with a proper call to the ActivatedRoute. Updated to `route.snapshot.params.application`